src/sick_mrs1000_communication.cpp

```diff
@@ -42,25 +42,41 @@ SickMrs1000Communication::SickMrs1000Communication(const std::string &hostname,
                                                    ScanAndCloudParser* parser)
 : SickTimCommonTcp(hostname, port, timelimit, parser),
   scan_and_cloud_parser_(parser),
-  cloud_pub_(nh_.advertise<sensor_msgs::PointCloud2>("cloud", 300)),
-  diagnosed_cloud_publisher_(cloud_pub_, diagnostics_,
-    diagnostic_updater::FrequencyStatusParam(&expectedFrequency_, &expectedFrequency_, 0.1, 10),
-    diagnostic_updater::TimeStampStatusParam(-1, 1.3 * 1.0/expectedFrequency_ - config_.time_offset)
-  )
+  cloud_pub_(nh_.advertise<sensor_msgs::PointCloud2>("cloud", 300))
 {
-  expectedFrequency_ = 12.5; // by data sheet 50 Hz, 4 x 12,5 Hz
+
+  updater_mrs_1000_ = new marble::DiagnosticUpdater("/"+namespace_+"/"+"cloud", nh_);
+  marble::diagnostics::FrequencyParams warning_freq_params;
+  warning_freq_params.min_frequency = config_.expected_fps - config_.fps_tolerance;
+  warning_freq_params.max_frequency = config_.expected_fps + config_.fps_tolerance;
+
+  marble::OutputDiagnosticParams output_cloud_params;
+  output_cloud_params.freq_warning_thresholds = warning_freq_params;
+  output_cloud_params.time_window_sec = 10.0;
+
+  output_cloud_diagnostic_ = new marble::OutputDiagnostic("/"+namespace_+"/"+"cloud", nh_, output_cloud_params);
+  output_cloud_diagnostic_->addToUpdater(updater_mrs_1000_);
+
+  generic_mrs_1000_diagnostic_ = new marble::GenericDiagnostic("datagram");
```

Comment:
Similarly, I think this generic element should be added to the `scan` updater?

Basically, I think we want both updaters to have the same set of elements added, except the scan updater should also have the scan output added, and the cloud updater should also have the cloud output added. Is that correct?

Reply:
Either way works but I didn't feel it was necessary to add generic diagnostics to a debug updater.

Reply:
Done